rcl/src/rcl/timer.c

```diff
@@ -254,17 +308,36 @@ rcl_timer_call(rcl_timer_t * timer)
     RCL_SET_ERROR_MSG("clock now returned negative time point value");
     return RCL_RET_ERROR;
   }
-  rcl_time_point_value_t previous_ns =
+  rcl_time_point_value_t previous_call_time =
     rcutils_atomic_exchange_int64_t(&timer->impl->last_call_time, now);
+  const int64_t time_since_last_call = now - previous_call_time;
+
   rcl_timer_callback_t typed_callback =
     (rcl_timer_callback_t)rcutils_atomic_load_uintptr_t(&timer->impl->callback);
 
   int64_t next_call_time = rcutils_atomic_load_int64_t(&timer->impl->next_call_time);
-  int64_t period = rcutils_atomic_load_uint64_t(&timer->impl->period);
+  const int64_t expected_time_since_last_call = next_call_time - previous_call_time;
+
+  const int64_t period = rcutils_atomic_load_uint64_t(&timer->impl->period);
+  int64_t period_correction = rcutils_atomic_load_int64_t(&timer->impl->period_correction);
```

Comment:
what's the point of the atomics here? (timer->impl->period_correction and timer->impl->period_jitter)